### PR TITLE
Feat: Update the Team Linter permissions test structure

### PR DESCRIPTION
**Pull request by ChrisSidebotham**
Refactor Team Permission Testing

# Overview/Summary

Replace this with a brief description of what this Pull Request fixes, changes, etc.

## This PR fixes/adds/changes/removes

1. Fixes the the testing logic for Github teams and associated permissions on the associated repositories

### Breaking Changes

1. Replaced function for performing test with new structure

## As part of this Pull Request I have

- [x] Read the Contribution Guide and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Verified-Modules/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Verified-Modules/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Verified-Modules/)
- [x] Ensured PR tests are passing
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by ChrisSidebotham** [comment]
@matebarabas

File: utilities/pipelines/sharedScripts/teamLinter/Test-AvmGitHubTeamPermission.ps1 [hunk 1]
```diff
@@ -0,0 +1,43 @@
+Function Test-AvmGitHubTeamPermission {
```
Comment (AlexanderSehr) [inline utilities/pipelines/sharedScripts/teamLinter/Test-AvmGitHubTeamPermission.ps1:1]: A synopsis would be great. For for later maintainability. Also

```suggestion
function Test-AvmGitHubTeamPermission {
```

😄

**Comment by ChrisSidebotham** [comment]
@matebarabas - Added additional owner existance check in group

File: utilities/pipelines/sharedScripts/teamLinter/Find-AvmGitHubTeamOwner.ps1 [hunk 2]
```diff
@@ -0,0 +1,35 @@
+function Find-AvmGitHubTeamOwner {
```
Comment (AlexanderSehr) [inline utilities/pipelines/sharedScripts/teamLinter/Find-AvmGitHubTeamOwner.ps1:23]: Would be good to have a synopsis

File: utilities/pipelines/sharedScripts/teamLinter/Test-AvmGitHubTeamPermission.ps1 [hunk 3]
```diff
@@ -1,4 +1,27 @@
...
+  The name of the GitHub Team.
+
+.PARAMETER RepoName
+  The name of the GitHub Repo.
+
+.PARAMETER ExpectedPermission
+  The expected permission for the team on the repo. `Admin | Write | Read | Triage | Maintain | None`
+
+.EXAMPLE
+  Test-AvmGitHubTeamPermission -Organization "myOrg" -TeamName "myTeam" -RepoName "myRepo" -ExpectedPermission "Admin"
```
Comment (AlexanderSehr) [inline utilities/pipelines/sharedScripts/teamLinter/Test-AvmGitHubTeamPermission.ps1:21]: Description missing

Comment (ChrisSidebotham) [reply utilities/pipelines/sharedScripts/teamLinter/Test-AvmGitHubTeamPermission.ps1:21]: Added

**Review by ChrisSidebotham — COMMENTED** [review]
Updated from Alex's comments

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**